### PR TITLE
[C API][COVERITY SCAN]Fix the TAINTED_SCALAR and DEADCODE in Coverity Scan

### PR DESCRIPTION
**Pull request by RICKIE777**
### Details:
 - *Fix the issues found in Coverity Scan*

### Tickets:
 - 87404

### CODE REVIEW

File: samples/c/hello_nv12_input_classification_ov/main.c [hunk 1]
```diff
@@ -104,10 +104,29 @@ void print_model_input_output_info(ov_model_t* model) {
  */
 
 bool is_supported_image_size(const char* size_str, size_t* width, size_t* height) {
-    char* p_end = NULL;
+    const char* _size = size_str;
     size_t _width = 0, _height = 0;
-    _width = strtoul(size_str, &p_end, 10);
-    _height = strtoul(p_end + 1, NULL, 10);
+    while (_size && *_size != 'x' && *_size != '\0') {
```
Comment (riverlijunjie) [inline samples/c/hello_nv12_input_classification_ov/main.c:109]: Why replace strtoul with below code block?

File: samples/c/hello_nv12_input_classification_ov/main.c [hunk 2]
```diff
@@ -118,11 +137,13 @@ bool is_supported_image_size(const char* size_str, size_t* width, size_t* height
             return false;
         }
     } else {
-        printf("Incorrect format of image size parameter, expected WIDTHxHEIGHT, "
-               "actual: %s\n",
-               size_str);
-        return false;
+        goto err;
```
Comment (riverlijunjie) [inline samples/c/hello_nv12_input_classification_ov/main.c:140]: else branch and goto error are not needed here.

File: samples/c/hello_nv12_input_classification_ov/main.c (hunk 1 above)
Comment (riverlijunjie) [reply samples/c/hello_nv12_input_classification_ov/main.c:109]: The problem comes that the returned width and height maybe be not assigned new value.

Comment (riverlijunjie) [reply samples/c/hello_nv12_input_classification_ov/main.c:109]: strtoul() will bring tainted argument , we shouldn't call strtol() function.

**Review by riverlijunjie — APPROVED** [review]

**Review by wangleis — APPROVED** [review]

**Pull request merged**